pkg/fleet-manager/backup_restore_migrate_shared.go

```diff
@@ -0,0 +1,306 @@
+// If every BackupStatusInCluster.Phase is marked as completed, it indicates that all backups have been successfully completed.
+func allBackupsCompleted(status backupapi.BackupStatus) bool {
+	for _, detail := range status.Details {
+		if detail.BackupStatusInCluster == nil || detail.BackupStatusInCluster.Phase != velerov1.BackupPhaseCompleted {
+			return false
+		}
+	}
+	return true
+}
+
+// deleteResourcesInClusters deletes instances of a Kubernetes resource based on the specified label key and value.
+// It iterates over all destination clusters and performs the deletion.
+// Parameters:
+// - ctx: context to carry out the API requests
+// - labelKey: the key of the label to filter the resources
+// - labelValue: the value of the label to filter the resources
+// - destinationClusters: a map containing information about the destination clusters
+// - objList: an empty instance of the resource list object to be filled with retrieved data. It should implement client.ObjectList.
+// Returns:
+// - error: if any error occurs during the deletion process
+// deleteResourcesInClusters deletes instances of a Kubernetes resource based on the specified label key and value.
+func deleteResourcesInClusters(ctx context.Context, namespace, labelKey string, labelValue string, destinationClusters map[ClusterKey]*fleetCluster, objList client.ObjectList) error {
+	// Log setup
+	log := ctrl.LoggerFrom(ctx)
+
+	// Iterate over each destination cluster
+	for clusterKey, clusterAccess := range destinationClusters {
+		// List the resources using the helper function
+		if err := listResourcesFromClusterClient(ctx, namespace, labelKey, labelValue, *clusterAccess, objList); err != nil {
+			log.Error(err, "Failed to list resources in cluster", "ClusterName", clusterKey.Name, "Namespace", namespace, "LabelKey", labelKey, "LabelValue", labelValue)
+			return err
+		}
+
+		// Extract Items using reflection
+		itemsValue := reflect.ValueOf(objList).Elem().FieldByName("Items")
+		if !itemsValue.IsValid() {
+			err := fmt.Errorf("failed to extract 'Items' from object list using reflection")
+			log.Error(err, "Reflection error")
+			return err
+		}
+
+		for i := 0; i < itemsValue.Len(); i++ {
+			item := itemsValue.Index(i).Addr().Interface().(client.Object)
+
+			clusterClient := clusterAccess.client.CtrlRuntimeClient()
+
+			if err := clusterClient.Delete(ctx, item); err != nil && !apierrors.IsNotFound(err) {
+				log.Error(err, "Failed to delete resource in cluster", "ResourceName", item.GetName(), "ResourceNamespace", item.GetNamespace(), "ClusterName", clusterKey.Name)
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+// if backup.Spec.Schedule is set, then it is ScheduleBackup. otherwise, it is regular/ont-time backup.
+func isScheduleBackup(backup *backupapi.Backup) bool {
+	return len(backup.Spec.Schedule) != 0
+}
+
+func generateVeleroInstanceLabel(createdByLabel, creatorName, fleetName string) map[string]string {
```

Comment:
nit: we have many duplicate calling these

Reply:
could you please clarify what you mean by 'many duplicate calling these'? Can you point to specific instances or suggest how I might address this?

Reply:
I mean we set labels in many different places

Reply:
refactor them in subsequent PRs.